_pytest/cacheprovider.py

```diff
@@ -147,11 +142,11 @@ def pytest_collection_modifyitems(self, session, config, items):
                     previously_failed.append(item)
                 else:
                     previously_passed.append(item)
-            if not previously_failed and previously_passed:
```

Comment:
I see, but unless I'm mistaken that is covered by the new test I added right?

I changed the condition to `if not previously_failed: return`, IOW don't skip anything if no collected item is part of the previous failures set.

So running `pytest test_b.py --lf` will only collect `test_b.py::*` tests, which means `previously_failed` will be empty and nothing will be skipped. At this point, if all tests from `test_b.py` pass, we don't lose the fact that `test_a.py::test_a` failed at some point in the past. If we execute `pytest test_a.py --lf`, now only `test_a.py::test_a` will execute, which is the point I'm trying to accomplish here with this PR.